beacon-chain/core/electra/deposits.go

```diff
@@ -386,8 +386,14 @@ func batchProcessNewPendingDeposits(ctx context.Context, state state.BeaconState
 		return errors.Wrap(err, "batch signature verification failed")
 	}
 
+	pubKeyMap := make(map[[48]byte]struct{}, len(pendingDeposits))
+	var found bool
+
 	// Process each deposit individually
 	for _, pendingDeposit := range pendingDeposits {
+		if _, found = pubKeyMap[[48]byte(pendingDeposit.PublicKey)]; !found {
+			pubKeyMap[[48]byte(pendingDeposit.PublicKey)] = struct{}{}
+		}
```

Comment:
nitpick: rather than declaring found outside the loop (which could lead to a bug in the future if someone tinkers), you could achieve the same goal at the appropriate scope with:
```
_, found := pubKeyMap[[48]byte(pendingDeposit.PublicKey)]
if !found {
    pubKeyMap[[48]byte(pendingDeposit.PublicKey)] = struct{}{}
}
``` (and delete `var found bool` outside the loop)